lmfit/minimizer.py

```diff
@@ -294,8 +294,10 @@ def penalty(self, params):
             objective function is an array, return the array sum-of-squares
         """
         r = self.__residual(params)
-        if isinstance(r, ndarray):
+        if isinstance(r, ndarray) and likelihood_fun is None:
```

Comment:
I seriously doubt that there are many use-cases where the overhead of lmfit plays a role, buf if proven otherwise i will revisit this.